Xamarin.Forms.Build.Tasks/XamlCTask.cs

```diff
@@ -236,6 +237,28 @@ public override bool Execute(IList<Exception> thrownExceptions)
 			return success;
 		}
 
+		static void ExpandMacro(Instruction instruction, OpCode opcode, object operand)
+		{
+			instruction.OpCode = opcode;
+			instruction.Operand = operand;
+		}
+
+		//this can be removed if/when https://github.com/jbevain/cecil/pull/307 is released in a nuget we consume
+		static void OptimizeLongs(MethodBody self)
+		{
+			var method = self.Method;
+			for (var i = 0; i < self.Instructions.Count; i++) {
+				var instruction = self.Instructions[i];
+				if (instruction.OpCode.Code != Code.Ldc_I8)
+					continue;
+				var l = (long)instruction.Operand;
+				if (l < int.MinValue || l > int.MaxValue)
+					continue;
+				ExpandMacro(instruction, OpCodes.Ldc_I4, unchecked((int)l));
+				self.Instructions.Insert(++i, Instruction.Create(OpCodes.Conv_I8));
+			}
```

Comment:
the best way to understand this, is to read the description of https://github.com/jbevain/cecil/pull/307